### PR TITLE
Add tag for fluentd image

### PR DESCRIPTION
**Pull request by dharmit**
#### Pull Request Checklist ####

- [x] Change does not remove any existing Images or Tags in the images-list file
- [x] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [ ] If updating an existing entry, verify the `SOURCE` is still accurate and upstream hasn't been migrated to a new regitry or repo (if they've migrated, a new repo request to EIO is needed to comply with the `SOURCE DESTINATION TAG` pattern)
- [x] New entries are in format `SOURCE DESTINATION TAG`
- [x] New entries are added to the correct section of the list (sorted lexicographically)
- [ ] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [x] New entries are licensed with Rancher favored licenses - Apache 2 and MIT - or approved licenses - as according to [CNCF approved licenses](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md).
- [ ] New entries, when used in Rancher or Rancher's provided charts, have their corresponding origin added in Rancher's images [origins](https://github.com/rancher/rancher/blob/release/v2.7/pkg/image/origins.go) file (must be added for all Rancher versions `>= v2.7`).
- [ ] Changes to scripting or CI config have been tested to the best of your ability

#### Linked Issues ####

Addresses part of https://github.com/rancherlabs/image-scanning/issues/1366
#### Final Checks after the PR is merged ####
- [ ] Confirm that you can pull the new images and tags from DockerHub

### CODE REVIEW

**Comment by dharmit** [comment]
@alexandreLamarre can you PTAL this PR and provide your feedback?

File: images-list [hunk 1]
```diff
@@ -222,6 +222,7 @@ ghcr.io/banzaicloud/fluentd rancher/mirrored-banzaicloud-fluentd v1.13.3-alpine-
 ghcr.io/banzaicloud/fluentd rancher/mirrored-banzaicloud-fluentd v1.14.4-alpine-2
 ghcr.io/banzaicloud/fluentd rancher/mirrored-banzaicloud-fluentd v1.14.5-alpine-1
 ghcr.io/banzaicloud/fluentd rancher/mirrored-banzaicloud-fluentd v1.14.6-alpine-5
+ghcr.io/kube-logging/fluentd rancher/mirrored-kube-logging-fluentd v1.16-ruby3.3-full-build.140
```
Comment (brandond) [inline images-list:225]: any commentary on why we're changing the tag format, and pulling from a different upstream?

Comment (brandond) [reply images-list:225]: Also, you'll need to create and link a ticket for EIO to create the `rancher/mirrored-kube-logging-fluentd` repository.

**Review by brandond — CHANGES_REQUESTED** [review]

File: images-list (hunk 1 above)
Comment (dharmit) [reply images-list:225]: > any commentary on why we're changing the tag format, and pulling from a different upstream?

The image is no longer maintained under ghcr.io/banzaicloud. The change in tag is a mere reflection of how ghcr.io/kube-logging tags the image. 

> Also, you'll need to create and link a ticket for EIO to create the `rancher/mirrored-kube-logging-fluentd` repository.

Ack. Thanks for the tip. I'm hoping for someone from the O&B team to validate if I've made the correct changes before I perform the next steps.

Comment (mallardduck) [reply images-list:225]: My main question on the new tag we're targeting is what one we should use to be "safest".
I [tracked down a default image tag](https://github.com/kube-logging/logging-operator/blob/ace8f631482fa031077e2ff6319a11a7d1174904/pkg/sdk/logging/api/v1beta1/logging_types.go#L175
) string to here:
> "v1.16-full"

However, for our use-case I think tagging to a specific build (as you've done) is apt per their Best Practice:  

> To ensure that subsequent builds don't break your production environment, you may want to pin your image to a specific build:
>     
>     ghcr.io/kube-logging/fluentd:VERSION-IMAGE-TYPE-build.BUILD_NUMBER

In theory the 3.3 ruby based images will have w/e benefits Ruby 3.3 brings over 3.2.
However if this isn't the default being used upstream, I wonder if that puts us at more risk by switching?

Comment (dharmit) [reply images-list:225]: > However, for our use-case I think tagging to a specific build (as you've done) is apt per their Best Practice:
> 
> > To ensure that subsequent builds don't break your production environment, you may want to pin your image to a specific build:
> > ```
> > ghcr.io/kube-logging/fluentd:VERSION-IMAGE-TYPE-build.BUILD_NUMBER
> > ```

This was the exact reason I went with a tag that looked kinda unreadable.



> if this isn't the default being used upstream, I wonder if that puts us at more risk by switching?

For the same reason that you mentioned above, IMO, we should use an image with build number.

Comment (mallardduck) [reply images-list:225]: Sorry to clarify - I think we're on the same page for targeting build specific tags.
I was less clear about what version "upstream" ships by default to users.
Are they using a Ruby 3.2 or Ruby 3.3 based image/fluentd build for `"v1.16-full"`?

My logic being that we probably want to use whatever version is most tested and used by end-users of upstream. That way we (in theory) have a lower risk of running into unexpected edge case issues. Or maybe I'm over thinking that part of things.

Maybe @joshmeranda has some ideas on this as he did a lot of hard work around the initial rebase to the new logging stuff.

**Comment by ericpromislow** [comment]
Doesn't look like you need another reviewer, but I can review that the ghrc.io image has no HIGH or CRITICAL CVEs

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Need to move the line starting with `ghcr.io/kube-logging/fluentd rancher/mirrored-kube-logging-fluentd` to the line after `ghcr.io/epinio/epinio-unpacker rancher/mirrored-epinio-epinio-unpacker v1.10.0` to pass the sort check.

File: images-list (hunk 1 above)
Comment (dharmit) [reply images-list:225]: > Or maybe I'm over thinking that part of things.

It's just our occupational hazard, I'd wager. 😉

**Comment by dharmit** [comment]
> Need to move the line starting with `ghcr.io/kube-logging/fluentd rancher/mirrored-kube-logging-fluentd` to the line after `ghcr.io/epinio/epinio-unpacker rancher/mirrored-epinio-epinio-unpacker v1.10.0` to pass the sort check.

Thanks, @ericpromislow. I thought images were grouped based on repository. Didn't know they were alphabetically sorted. :)

**Comment by brandond** [comment]
lgtm; still need the new target repo created by EIO

**Comment by dharmit** [comment]
> lgtm; still need the new target repo created by EIO

@brandond done. Can you PTAL when you have a chance?

**Review by mallardduck — APPROVED** [review]
:shipit:

**Review by brandond — APPROVED** [review]

**Pull request merged**